earthsim/__init__.py

```diff
@@ -0,0 +1,45 @@
+import param
+
+def params_from_kwargs(**kwargs):
+    """
+    Utility to promote keywords with literal values to the appropriate
+    parameter type with the specified default value unless the value is
+    already a parameter.
+    """
+    params = {}
+    for k, v in kwargs.items():
+        kws = dict(default=v)
+        if isinstance(v, param.Parameter):
+            params[k] = v
+        elif isinstance(v, bool):
+            params[k] = param.Boolean(**kws)
+        elif isinstance(v, int):
+            params[k] = param.Integer(**kws)
+        elif isinstance(v, float):
+            params[k] = param.Number(**kws)
+        elif isinstance(v, str):
+            params[k] = param.String(**kws)
+        elif isinstance(v, dict):
+            params[k] = param.Dict(**kws)
+        elif isinstance(v, tuple):
+            params[k] = param.Tuple(**kws)
+        elif isinstance(v, list):
+            params[k] = param.List(**kws)
+        elif isinstance(v, np.ndarray):
+            params[k] = param.Array(**kws)
```

Comment:
This is code I've used in at least three separate contexts, once in another project and once in holoviews itself. I would like to see it moved to param instead of duplicating this code.

Reply:
I'm happy for it to move to param, though if it's done there, it will need a good description mentioning the limitations (i.e. that only a few Parameter types will ever be supported in this way).

Reply:
I'll make a PR to param for this code but I don't think this PR should wait on that. Once this is in param, I can update earthsim to import it accordingly.